### PR TITLE
Immutable VersionInfo

### LINKED ISSUE #96: VersionInfo shouldn't be mutable
From https://github.com/k-bx/python-semver/pull/93#discussion_r191442817

We should add tests to ensure immutability of VersionInfo

### PR DESCRIPTION
**Pull request by scls19fr**
Closes #96

### CODE REVIEW

**Comment by scls19fr** [comment]
I don't understand why CI is broken with pypy and Python 2.7.

Any idea?

File: semver.py [hunk 1]
```diff
@@ -82,13 +82,40 @@ class VersionInfo:
...
+
+    @property
+    def prerelease(self):
+        return self._prerelease
+
+    @property
+    def build(self):
+        return self._build
+
+    def __setattr__(self, name, value):
```
Comment (ppkt) [inline semver.py]: I think this function is unnecessary

File: test_semver.py [hunk 2]
```diff
@@ -414,3 +414,42 @@ def test_parse_method_for_version_info():
     s_version = "1.2.3-alpha.1.2+build.11.e0f985a"
     v = VersionInfo.parse(s_version)
     assert str(v) == s_version
+
+
+def test_immutable():
```
Comment (ppkt) [inline test_semver.py]: Could you also add a test to check if it's possible to add new attribute? Something like this:
```
try:
    v.some_attribute = 1
except:
    ...
```

File: semver.py (hunk 1 above)
Comment (scls19fr) [reply semver.py]: I thought also that __setattr__ wasn't necessary...
but not with Python 2.7 and Pypy

Comment (kxepal) [reply semver.py]: Why do we want to have private attributes be mutable?

File: semver.py [hunk 3]
```diff
@@ -82,13 +82,40 @@ class VersionInfo:
...
+        if name not in self.__slots__:
+            raise AttributeError
```
Comment (kxepal) [inline semver.py]: Even if so, it's better to pass name to AttributeError to avoid any confusions.

File: semver.py (hunk 1 above)
Comment (scls19fr) [reply semver.py]: You are right ... 
[This build](https://travis-ci.org/k-bx/python-semver/builds/386245547?utm_source=github_status&utm_medium=notification) shows that  `__setattr__` is not required.

Although what is required for Python 2.7 and Pypi is inheriting VersionInfo from `object`.

**Pull request merged**